app/app.go

```diff
@@ -502,11 +573,31 @@ func NewOKExChainApp(
 	app.TransferKeeper = *app.TransferKeeper.SetHooks(erc20.NewIBCTransferHooks(app.Erc20Keeper))
 	transferModule := ibctransfer.NewAppModule(app.TransferKeeper, codecProxy)
 
+	left := common.NewDisaleProxyMiddleware()
+	middle := ibctransfer.NewIBCModule(app.TransferKeeper, transferModule)
+	right := ibcfee.NewIBCMiddleware(middle, app.IBCFeeKeeper)
+	transferStack := ibcporttypes.NewFallThroughMiddleware(left,
+		ibccommon.DefaultFactory(tmtypes.HigherThanVenus4, ibc.IBCV4, right),
+		ibccommon.DefaultFactory(tmtypes.HigherThanVenus1, ibc.IBCV2, middle))
+
 	// Create static IBC router, add transfer route, then set and seal it
 	ibcRouter := ibcporttypes.NewRouter()
-	ibcRouter.AddRoute(ibctransfertypes.ModuleName, transferModule)
+
+	var icaControllerStack ibcporttypes.IBCModule
+	icaMauthIBCModule := icamauth.NewIBCModule(app.ICAMauthKeeper)
```

Comment:
why not use ' icaMauthIBCModule := icamauth.NewIBCModule(app.ICAMauthKeeper)
	icaControllerStack := icacontroller.NewIBCMiddleware(icaMauthIBCModule, app.ICAControllerKeeper) ' to replace this codes